README.md

```diff
@@ -200,17 +253,31 @@ If raising a Java `Exception`, throw new instance of Java `Exception` class as d
 
     throw $.newNullPointerException("Cannot copy to/from a null array.");
 
+If you need implement a native method with async JS calls, the following steps are required:
+
+1. Add the method to the `yieldMap` in jit/analyze.ts
+2. Use `asyncImpl` in override.js to return the asnyc value with a `Promise`.
+
+The `asyncImpl` is optional if part of the code doesn't make async calls, the method could return value as regular native method and the VM will handle it properly.
+
+e.g:
+
+    Native["java/lang/Thread.sleep.(J)V"] = function(delay) {
+        asyncImpl("V", new Promise(function(resolve, reject) {
+            window.setTimeout(resolve, delay.toNumber());
+        }));
+    };
```

Comment:
@mykmelez Fixed in  45c0837. I add an example of partly asynchronous native methods as well.